extra/generic-init.d/celeryd

```diff
@@ -39,11 +39,14 @@ fi
 
 
 # Can be a runlevel symlink (e.g. S02celeryd)
-if [[ `dirname $0` == /etc/rc*.d ]]; then
+set +e
```

Comment:
`+e` is the default, but you are setting it to -e later?

If the set -e is propagated from something else calling the script then it would be better to
wrap it in a function:

    origin_is_runlevel_dir () {
        set +e
        dirname $0 | grep -q "/etc/rc.\.d"
        echo $?
    }

    if [ $(origin_is_runlevel_dir) -eq 0 ]; then